apiserver/machinemanager/machinemanager.go

```diff
@@ -198,6 +229,12 @@ func (mm *MachineManagerAPI) destroyMachine(args params.Entities, force bool) (p
 		if err != nil {
 			return nil, err
 		}
+		if keep {
+			logger.Infof("destroy machine %v but keep instance", machineTag.Id())
+			if err := machine.SetKeepInstance(keep); err != nil {
```

Comment:
This is one potential call site, albeit the only one right now. However, why hard code an API in a lower layer just because  right now we only set the value to true?